tests/testthat/setup-testing_depth.R

```diff
@@ -0,0 +1,56 @@
+#' Returns testing depth set by an environmental variable.
+#'
+#' @details Looks for the option `TESTING_DEPTH` first, if not set,
+#' takes the system environmental variable `TESTING_DEPTH`. If neither
+#' is set, then returns 3 by default. If the value of `TESTING_DEPTH`
+#' is not a scalar numeric, then returns 3.
+#'
+#' @return `numeric(1)` the testing depth.
+#'
+testing_depth <- function() {
+  testing_depth <- getOption("TESTING_DEPTH")
+  if (is.null(testing_depth)) {
+    testing_depth <- Sys.getenv("TESTING_DEPTH")
+  }
+
+  testing_depth <- tryCatch(
+    as.numeric(testing_depth),
+    error = function(error) 3,
+    warning = function(warning) 3
+  )
+
+  if (length(testing_depth) != 1 || is.na(testing_depth)) {
+    testing_depth <- 3
+  }
```

Comment:
If you do not allow `testing_depth` to be `""` (when `Sys.getenv()` returns `""`) https://github.com/insightsengineering/teal/pull/1161/files#r1531798495 then `as.numeric("")` doesn't make it `NA`
```suggestion
  testing_depth <- testing_depth[1]
```

Reply:
an unset value solves this.

Reply:
Reference commit 84568ec81c453e3cb3a37b161661e3a3135f92ab